### ISSUE TITLE
even with a custom bundle gemfile it uses the local .tool-version for setting up ruby

### ISSUE THREAD

**Issue by estiens** [Shopify/vscode-ruby-lsp#758]
### Operating System

Mac

### Ruby version

3.1.1

### Project has a bundle

- [X] Has bundle

### Ruby version manager being used

asdf

### Description

I have a number of projects using Ruby 2.7.X
I have ruby-lsp installed in a custom directory using ruby 3.1.1
I do not have ruby-lsp in that gemfile
I did run ruby-lsp to create a .ruby-lsp directory that has a Gemfile and Gemfile.lock
I put a .tool-versions file in that directory

However, I still get "Failed to activate asdf environment, requires Ruby 3"
expected behavior - if using a custom bundleGemfile, uses whatever ruby version is there to run and doesn't worry about the local projects .tool-version

**Comment by alextakitani** [comment]
I have the same problem. 

I could have it working by changing in settings.json:

"rubyLsp.customRubyCommand": "/home/alex/.asdf/installs/ruby/3.2.0/bin/ruby",
"rubyLsp.rubyVersionManager": "custom",

but now formating does not work.

**Comment by vinistock** [comment]
Yeah, definitely. If you clone the [server](https://github.com/Shopify/ruby-lsp) and open the project, the extension will use that local clone to run.

You can add `warn` statements in the code to figure out what's going wrong. Do print anything to `STDOUT` because that will mess up the communication between the LSP and the editor and you'll end up in an infinite loop.

First, you could try adding a warn [here](https://github.com/Shopify/ruby-lsp/blob/main/lib/ruby_lsp/executor.rb#L317) to check if we're identifying the right formatter or not. If it's correct, it should be `rubocop`.
